neuralmonkey/encoders/attentive.py

```diff
@@ -0,0 +1,25 @@
+# pylint: disable=too-few-public-methods
+class Attentive(object):
+    def __init__(self, attention_type, **kwargs):
+        self._attention_type = attention_type
+        self._attention_kwargs = kwargs
+
+        assert hasattr(self, "name")
+        assert hasattr(self, "_padding")
+        assert hasattr(self, "_attention_tensor")
```

Comment:
Jako že `Decoder._collect_attention_objects` by ještě na těch enkodérech volalo něco jako `if encoder.supports_attention:  encoder.create_attention_graph()`, který by vracelo `Attentive` nebo `Attention` objekt.

@jlibovicky šlo by to takhle? Změnila by se tam ta hierarchie, že místo encoder is_a attentive by to bylo encoder has_a attention

Taky to můžeme nechat na pak, aby se na tenhle pull request zbytečně nenabaloval další a další kód.